lib/identity_pki/token.ex

```diff
@@ -0,0 +1,54 @@
+defmodule IdentityPki.Token do
+  def box(data, opts \\ []) do
+    expires_at = Keyword.get(opts, :expires_at)
+    key = Keyword.get_lazy(opts, :key, &secret_key/0)
+
+    data =
+      if expires_at do
+        Map.put(data, :expires_at, expires_at)
+      else
+        data
+      end
+
+    json =
+      Map.put(data, :r1, random_bytes(8))
+      |> Map.put(:r2, random_bytes(8))
+      |> Jason.encode!()
+
+    encrypt_text(json, key)
+  end
+
+  def open(token, key \\ secret_key()) do
+    # TODO: check expires_at
+    # TODO: check HMAC
+    [text, iv, tag] =
+      String.split(token, "--")
+      |> IO.inspect(label: "SPLIT TOKEN")
+      |> Enum.map(&Base.decode64!/1)
+
+    :crypto.crypto_one_time_aead(:aes_256_gcm, key, iv, text, "", tag, false)
+    |> Jason.decode!()
```

Comment:
omg they called the JSON library Jason 😂 😭

Reply:
It's great!